### DISCUSSION TITLE
wechaty是否可以获取到群成员的手机号？

### DISCUSSION THREAD

**Discussion by weishaogeng512** [wechaty/wechaty#2401] [Q&A] [answered]
我做了一个小程序，客户登录小程序的时候只能获取到他的手机号，没有办法获取到客户的微信号。

我用wechaty做了群签到的功能，签到之后目前能获取到客户的wxid，也就是微信号。我查看wechaty的doc说明是能获取到phone的，但是我调取之后，返回的是空值。

所以想请教一下wechaty是否可以获取到群成员的手机号?

如果可以获取，我的小程序和群签到功能就能串联起来了。

**Comment by huan** [comment, marked as answer]
This feature will depend on the Puppet Provider implementation.

If the Puppet Provider provides this payload data, then Wechaty will be able to read it.

Learn more about the Puppet Provider from https://wechaty.js.org/docs/puppet-providers/

**Reply by weishaogeng512** [reply]
OK ,thanks!

**Reply by huan** [reply]
Your are welcome!
